Assets/BossRoom/Scripts/Client/Game/State/ClientCharSelectState.cs

```diff
@@ -395,6 +395,11 @@ public void OnPlayerExit()
         {
             // Player is leaving the group
             // first disconnect then return to menu
+            if (IsServer)
+            {
+                NetworkObject networkObject = gameObject.GetComponent<NetworkObject>();
+                networkObject.Despawn();
```

Comment:
So I have been mulling over this issue...need to see if there is a Jira ticket already open for it... but it definitely has to do with the NetworkManager shutdown process.  
I think something like this in NetworkSpawnManager should do the trick:
```
internal void DespawnNetworkObjects()
{
    var networkObjects = UnityEngine.Object.FindObjectsOfType<NetworkObject>();

    for (int i = 0; i < networkObjects.Length; i++)
    {
        if (networkObjects[i].NetworkManager == NetworkManager)
        {
            if (NetworkManager.PrefabHandler.ContainsHandler(networkObjects[i]))
            {
                NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(networkObjects[i]);
                // Leave destruction up to the handler
                OnDespawnObject(networkObjects[i], false);
            }
            else
            {
                // If it is an in-scene placed NetworkObject then just despawn
                // and let it be destroyed when the scene is unloaded.  Otherwise,
                // despawn and destroy it.
                var shouldDestroy = !(networkObjects[i].IsSceneObject != null
                            && networkObjects[i].IsSceneObject.Value);

                OnDespawnObject(networkObjects[i], shouldDestroy);
            }
        }
    }
}
```

Then in the NetworkManager.Shutdown:
```
if (SpawnManager != null)
{
    //SpawnManager.DestroyNonSceneObjects();
    SpawnManager.DespawnNetworkObjects();
    SpawnManager.ServerResetShudownStateForSceneObjects();

    SpawnManager = null;
}
```

This will fix the issues you are experiencing.  I already tested the above code adding an override for OnNetworkDespawn in the NetworkPrefabHandlerAdditive class and after loading an additive scene with that component and then exiting back to the main menu (i.e. immediately shutting down) it does invoke the OnNetworkDespawn method:
![image](https://user-images.githubusercontent.com/73188597/139969175-dcf435fa-7587-415c-80de-8e03ba7e2540.png)

If I uncomment the DestroyNonSceneObjects() and comment out the DespawnNetworkObjects() in the NetworkManager.Shutdown it does not invoke.

Will look to see if there is an already existing Jira Ticket and if not I will open one up and include the above in a PR for this issue along with tests to validate ( which I need to noodle over that a little ).  

While I do that, it would help if someone could apply the above suggested fix to a version of Boss Room that doesn't have this fix (or revert back the singleton reference)  and see if this indeed fixes your this issue?

Reply:
[PR-1390](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1390) has several updates (including an update to the above code) that are required in many cases but some need to be discussed to determine the best approach... but if you wanted to see if Boss Room would work...test it with the PR's branch.

I will also test it on my end sometime tomorrow.

Reply:
I just tested with the PR branch and it seems to work, I cannot reproduce the bug.

Reply:
Need to wait until @MFatihMAR returns from vacation since some of the adjustments made in PR-1390 makes adjustments to some areas he is the T.O. of and so the earliest it will get merged will be sometime late next week.  
Not sure when the next backport is slated, but hopefully 1390 will make it out in the next one.